### ISSUE TITLE
Make --needed such that unnecessary trials with a batch can be skipped

### ISSUE THREAD

**Issue by thomsentner** [guildai/guildai#251]
Hi,

I'm running into a problem with `guild run --needed`. If I initiate a batch run, `--needed` only checks whether the batch run was executed previously. If it doesn't find a matching batch run, it will run all trials of the batch. It won't check if any of these individual trials was already ran standalone.

**Comment by gar1t** [comment]
My apologies - I missed a number of issues in Nov including this one! In this case you need to include the `--needed` option in the step spec as well as setting `isolate-runs` to `no` like this:

```
op: {}
pipeline:
  steps:
    - run: op --needed
      isolate-runs: no
```

I don't like this interface - it's wonky. But this is how to do it in 0.7.x. 0.8 will have a cleaner implementation of pipelines.

Sorry again for the late reply!

**Comment by thomsentner** [comment]
No worries on the late reply.

Just to confirm, this would only be possible by editing the guild file right; your suggestion is not possible in a bash command?

**Comment by gar1t** [comment]
Yes that's right - this is the `run` entry for each step in the Guild file. The `--needed` flag on the command line just applies to the top-level operation and doesn't apply to its steps.
